test/framework/src/main/java/org/opensearch/cluster/coordination/MockSinglePrioritizingExecutor.java

```diff
@@ -44,25 +46,30 @@
 public class MockSinglePrioritizingExecutor extends PrioritizedOpenSearchThreadPoolExecutor {
 
     public MockSinglePrioritizingExecutor(String name, DeterministicTaskQueue deterministicTaskQueue, ThreadPool threadPool) {
-        super(name, 0, 1, 0L, TimeUnit.MILLISECONDS, r -> new Thread() {
-            @Override
-            public void start() {
-                deterministicTaskQueue.scheduleNow(new Runnable() {
-                    @Override
-                    public void run() {
-                        try {
-                            r.run();
-                        } catch (KillWorkerError kwe) {
-                            // hacks everywhere
-                        }
+        super(name, 0, 1, 0L, TimeUnit.MILLISECONDS, r -> {
+            // This executor used to override Thread::start method so the actual runnable is
+            // being scheduled in the scope of current thread of execution. In JDK-19, the Thread::start
+            // is not called anymore (https://bugs.openjdk.org/browse/JDK-8292027) and there is no
+            // suitable option to alter the executor's behavior in the similar way. The closest we
+            // could get to is to schedule the runnable once the ThreadFactory is being asked to
+            // allocate the new thread.
```

Comment:
This feels like a problem. How did you come up with a workaround and what do folks in  https://bugs.openjdk.org/browse/JDK-8292027 think about it? Maybe add a comment there?

Reply:
@dblock it does, and the fact the issue is still open and no alternative mechanism provided, it will be the probelm for quite some time. 

I tried several approaches, the most straightforward one is to not override `start` and pass wrapped runnable, it turned out to be exceptionally problematic: from single threaded behaviour we are getting multi-threaded one (`DeterministicTaskQueue` is absolutely not suitable for that).

The idea behind this change: we know that the new thread is going to be allocated once thread pool decides that new worker is needed. We cannot override or alter the `addWorker` behaviour but we could "simulate" `Thread::start` at the moment new `Thread` is being created.

This implementation is laser target to the way `MockSinglePrioritizingExecutor` / `DeterministicTaskQueue` are used in the tests, I would not recommend it to be used anywhere else.

Reply:
Thanks for the detailed explanation.